tests/test_pay.py

```diff
@@ -1873,131 +1873,144 @@ def test_setchannelfee_usage(node_factory, bitcoind):
     DEF_BASE = 10
     DEF_BASE_MSAT = Millisatoshi(DEF_BASE)
     DEF_PPM = 100
+    # Minus reserves
+    MAX_HTLC = Millisatoshi(int(FUNDAMOUNT * 0.98))
 
     l1, l2, l3 = node_factory.get_nodes(3,
                                         opts={'fee-base': DEF_BASE, 'fee-per-satoshi': DEF_PPM})
     node_factory.join_nodes([l1, l2])
     l1.rpc.connect(l3.info['id'], 'localhost', l3.port)
 
-    def channel_get_fees(scid):
+    def channel_get_config(scid):
         return l1.db.query(
-            'SELECT feerate_base, feerate_ppm FROM channels '
+            'SELECT feerate_base, feerate_ppm, htlc_maximum_msat FROM channels '
             'WHERE short_channel_id=\'{}\';'.format(scid))
 
     # get short channel id
     scid = l1.get_channel_scid(l2)
 
     # feerates should be init with global config
-    db_fees = l1.db_query('SELECT feerate_base, feerate_ppm FROM channels;')
+    db_fees = l1.db_query('SELECT feerate_base, feerate_ppm, htlc_maximum_msat FROM channels;')
     assert(db_fees[0]['feerate_base'] == DEF_BASE)
     assert(db_fees[0]['feerate_ppm'] == DEF_PPM)
+    # This will be the capacity - reserves:
+    assert(db_fees[0]['htlc_maximum_msat'] == MAX_HTLC)
```

Comment:
This line fails.

First issue here is we compare a number int (msat) with msat value (BUT shifted by 1000, since we did `Millisatoshi(int(FUNDAMOUNT * 0.99)))` which should be `Millisatoshi(int(FUNDAMOUNT * 1000 * 0.99)))`. To prevent that we may change FUNDAMOUNT to FUNDAMOUNT_SAT everywhere. This seems also messed up in other locations and tests in this file: `test_setchannel_routing`, `test_setchannel_restart`, ... search for `FUNDAMOUNT`, don't know why they not fail or why we haven't seen earlier.

Second issue: The DB thinks we can send 99% of the channel capacity (which seems correct to me). Where we test for 98%. Since this is a pristine new channel, with l2 having 0msat, we should be able to send 99% just so we keep our own reserve filled. If l2 decides later to send amounts back he can then only do 98% of the FUNDAMOUNT.

```py
>       assert(db_fees[0]['htlc_maximum_msat'] == MAX_HTLC)
E       assert 990000000 == 980000msat
E         +990000000
E         -980000msat
```

Reply:
Oh yes, I pushed this up early.  I've reworked the tests now.